packages/ui/components/form/selectimproved/components/Select.tsx

```diff
@@ -0,0 +1,260 @@
+import React, { useCallback, useEffect, useRef, useState } from "react";
+
+import { classNames as cn } from "@calcom/lib";
+import { useKeyPress } from "@calcom/lib/hooks/useKeyPress";
+import useOnClickOutside from "@calcom/lib/hooks/useOnclickOutside";
+
+import { FiX, FiChevronDown } from "../../../icon";
+import Options from "./Options";
+import SearchInput from "./SearchInput";
+import SelectProvider from "./SelectProvider";
+import { Option } from "./type";
+
+interface SelectProps<T extends Option> {
+  options: T[];
+  selectedItems: T | T[] | null;
+  onChange: (value?: Option | Option[] | null) => void;
+  placeholder?: string;
+  isMultiple?: boolean;
+  isClearable?: boolean;
+  isSearchable?: boolean;
+  isDisabled?: boolean;
+  loading?: boolean;
+  menuIsOpen?: boolean;
+  searchInputPlaceholder?: string;
+  noOptionsMessage?: string;
+  classNames?: {
+    menuButton?: ({ isDisabled }: { isDisabled: boolean }) => string;
+    menu?: string;
+    tagItem?: ({ isDisabled }: { isDisabled: boolean }) => string;
+    tagItemText?: string;
+    tagItemIconContainer?: string;
+    tagItemIcon?: string;
+    list?: string;
+    listGroupLabel?: string;
+    listItem?: ({ isSelected }: { isSelected: boolean }) => string;
+    listDisabledItem?: string;
+    ChevronIcon?: ({ open }: { open: boolean }) => string;
+    searchContainer?: string;
+    searchBox?: string;
+    searchIcon?: string;
+    closeIcon?: string;
+  };
+}
+
+function Select<T extends Option>({
+  options = [],
+  selectedItems = null,
+  onChange,
+  placeholder = "Select...",
+  searchInputPlaceholder = "Search...",
+  isMultiple = false,
+  isClearable = false,
+  isSearchable = false,
+  isDisabled = false,
+  menuIsOpen = false,
+  classNames,
+}: SelectProps<T>) {
+  const [open, setOpen] = useState<boolean>(menuIsOpen);
+  const [inputValue, setInputValue] = useState<string>("");
+  const ref = useRef<HTMLDivElement>(null);
+  const searchBoxRef = useRef<HTMLInputElement>(null);
+  const escapePressed = useKeyPress("Escape");
+  const isMultipleValue = Array.isArray(selectedItems) && isMultiple;
+
+  const toggle = useCallback(() => {
+    if (!isDisabled) {
+      setOpen(!open);
+    }
+  }, [isDisabled, open]);
+
+  const closeDropDown = useCallback(() => {
+    if (open) setOpen(false);
+  }, [open]);
+
+  useOnClickOutside(ref, () => {
+    closeDropDown();
+  });
+
+  useEffect(() => {
+    if (escapePressed) {
+      closeDropDown();
+    }
+  }, [escapePressed, closeDropDown]);
```

Comment:
Maybe something similar as above 

```suggestion
   useOnEscKeypress(ref, () => {
     closeDropDown();
   });```